lib/containers/runtime.pm

```diff
@@ -0,0 +1,142 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::runtime;
+use Mojo::Base -base;
+use testapi;
+use Test::Assert 'assert_equals';
+
+has runtime => undef;
+
+sub _rt_assert_script {
+    my ($self, $cmd, @args) = @_;
+    assert_script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_run {
+    my ($self, $cmd, @args) = @_;
+    return script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_output {
+    my ($self, $cmd, @args) = @_;
+    return script_output($self->runtime . " " . $cmd, @args);
+}
+
+=head2 build
+
+Build a container after visit C<dockerfile_path>.
+C<dockerfile_path> is the directory with the Dockerfile.
+C<container_tag> will be the name of the container.
+
+=cut
+sub build {
+    my ($self, $dockerfile_path, $container_tag) = @_;
+    die 'wrong number of arguments' if @_ < 3;
+    assert_script_run "cd $dockerfile_path";
+    $self->_rt_assert_script("build -t $container_tag .");
+    record_info "$container_tag created";
+}
+
+=head2 up
+
+Run a container.
+If C<daemon> is enabled then container will run in the daemon mode. Otherwise will be in the 
```

Comment:
I assume whole idea of this file to abstract away from docker/podman , right ? as far as I know podman does not have daemon, only docker have it . if I am right than I am little bit puzzled with what written here

Reply:
https://podman.readthedocs.io/en/latest/markdown/podman-run.1.html the correct terminology i guess is `detached`. When you run the container *as a daemon* means it is not in the interactive mode. i will try to rephrase it.

Reply:
ok now I see where I misunderstood you - I thought you speaking about docker daemon but you just meant running container as daemon . but again from what I briefly read this only relevant to podman . there is no such thing as `daemon mode ` for docker , right ?

Reply:
> https://podman.readthedocs.io/en/latest/markdown/podman-run.1.html the correct terminology i guess is `detached`. When you run the container _as a daemon_ means it is not in the interactive mode. i will try to rephrase it.

What about let's be detached as long as it is not specified otherwise? :)

Reply:
s/daemon/detached/ in the description